src/components/error-boundary/error-boundary.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactNode, ErrorInfo } from 'react';
+import { formatMessage } from '../../utility/errors';
+import { AlertDialog, AlertDialogProps } from '../alert-dialog/alert-dialog';
+
+
+export interface ErrorDialogProps extends AlertDialogProps {
+  errorInfo?: ErrorInfo | null,
+}
+
+export const ErrorDialog = (props: ErrorDialogProps) => {
+  return AlertDialog(props);
+};
+
+
+export interface ErrorBoundaryProps {
+  children?: ReactNode,
+  errorTitle?: string,
+  errorMessage?: string,
+  errorDialogProps: ErrorDialogProps,
+  setErrorDialogProps: (props: ErrorDialogProps) => void,
```

Comment:
I'm not sure if passing error data through in the props makes much sense. Right now the Roster and App components are declaring `errorDialogProps` and `setErrorDialogProps`, then passing them into the ErrorBoundary but not actually doing anything with them.

Shouldn't the error boundary itself contain that stuff, with the error that comes through populating them? I don't think the ErrorBoundary component should ever receive any props, since it's potentially higher up in the hierarchy than the component that threw the error.

Putting that stuff in the ErrorBoundary component will look a little different than the other components, since it's a class component and they're functional ones using hooks, but you should be able to do all of the same stuff via its internal state without passing it through in the props.